presto-main/src/main/java/com/facebook/presto/execution/resourceGroups/InternalResourceGroupManager.java

```diff
@@ -72,7 +75,9 @@
     private static final File RESOURCE_GROUPS_CONFIGURATION = new File("etc/resource-groups.properties");
     private static final String CONFIGURATION_MANAGER_PROPERTY_NAME = "resource-groups.configuration-manager";
 
+    private final ExecutorService refreshResourceGroupExecutor = newCachedThreadPool(threadsNamed("query-scheduler-%s"));
```

Comment:
this executor is to run the query. It is being passed to InternalResourceGroupManager#submit method when it's called. But given now we are creating InternalResourceGroup from the InternalResourceGroupManager, we need to provide executor from within.

Reply:
In this case let's use the same executor and use dependency injection to use the same executor service from SqlQueryManager, rather than create a duplicate with an identically named thread factory.